File: class_1/multi_threads.go
package main

import(
	"time"
	"fmt"
)

func thread(name string, delay time.Duration){
	fmt.Println(name,"run")
	time.Sleep(delay * time.Second)
	fmt.Println(name,"exit")
}

func main(){
	for i := 0; i < 1000; i++{
		go thread(fmt.Sprintf("Thread-%d", i), 10000)
	}
	time.Sleep(1000000*time.Second)
}

File: class_1/multi_threads.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
 
import thread
import time
 
# 为线程定义一个函数
def print_time( threadName, delay):
   print "Thread-%d"%(threadName) + " run"
   time.sleep(delay)
   print threadName + " exit"
 
# 创建两个线程
try:
   for i in range(0, 999):
      thread.start_new_thread( print_time, (i, 100000, ) )
except:
   print "Error: unable to start thread"
 

time.sleep(1000000000)
